src/utilities/findBreakingChanges.js

```diff
@@ -676,3 +677,21 @@ export function findInterfacesAddedToObjectTypes(
   });
   return interfacesAddedToObjectTypes;
 }
+
+export function findRemovedDirectives(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema,
+): Array<BreakingChange> {
+  const removedDirectives = [];
+
+  oldSchema.getDirectives().forEach(directive => {
+    if (!newSchema.getDirective(directive.name)) {
```

Comment:
This I believe makes this O(n^2): https://github.com/graphql/graphql-js/blob/master/src/type/schema.js#L206

A more efficient version, O(nlogn):

```
const newSchemaDirectiveNames = new Set(oldSchema.getDirectives().map(directive => directive.name));
oldSchema.getDirectives().forEach(directive => {
  if (!newSchemaDirectiveNames.has(directive.name) {
    ...
  }
}
```

Reply:
👍  thanks @mjmahone !

Reply:
update